src/Entity/Product.php

```diff
@@ -106,6 +159,11 @@ public function setSlug(string $slug): self
         return $this;
     }
 
+    public function computeSlug(SluggerInterface $slugger)
+    {
+        $this->slug = (string) $slugger->slug((string) $this->name)->lower();
+    }
+
```

Comment:
This method should not be in the entity and does not present much interest. You should move it as a private method of your Listener

Reply:
Done ! :)